supervisor/loggers.py

```diff
@@ -387,10 +396,15 @@ def handle_file(logger, filename, fmt, rotating=False, maxbytes=0, backups=0):
         if rotating is False:
             handler = FileHandler(filename)
         else:
-            handler = RotatingFileHandler(filename, 'a', maxbytes, backups)
+            handler = RotatingFileHandler(filename, 'a', maxbytes, backups,
+                                          compress)
 
     handler.setFormat(fmt)
     handler.setLevel(logger.level)
     logger.addHandler(handler)
 
     return logger
+
+def handle_compression(sfn, dfn):
+    with tarfile.open(dfn, "w:gz") as archive:
+        archive.add(sfn, arcname=os.path.basename(sfn))
```

Comment:
It may be possible to do this inside `supervisord` in a non-blocking way but I'm not sure how at the moment.  Doing it in an [eventlistener subprocess](http://supervisord.org/events.html#events) may be more straightforward and has no risk of blocking `supervisord` during compression.  If you haven't written an eventlistener before, there's some examples in the [Superlance](https://github.com/supervisor/superlance) package. 

An eventlistener can get info about all processes using the `getAllProcessInfo()` method of the XML-RPC interface, and this includes the path to the current log files of each process.

The eventlistener wouldn't have a way to know when log rotation occurred.  This could be addressed by adding a new event to `supervisord` to indicate when a rotation has occurred.  It may not be necessary to do this, the eventlistener could just glob every so often on a `TICK_` event.  If a `logfile.2` exists but no `logfile.1.gz`, time to gzip.